### DISCUSSION TITLE
Enhancing `Khoj` with Next.js

### DISCUSSION THREAD

**Discussion by Raghav-45** [khoj-ai/khoj#743] [Q&A]
Recently I've been working on a chat project and I'm eager to contribute it to **_Khoj_**. However, there's a slight hiccup – my UI is built with ReactJS, Tailwind CSS while **_Khoj_** is using Django. To ensure the compatibility of the project, I propose transitioning towards `Next.js`, which is better and future-proof 🚀

Here are some screenshots showcasing my UI:

![Screenshot (169)](https://github.com/khoj-ai/khoj/assets/77260113/d077e574-9973-4eaf-8889-c81b0245154d)
![Screenshot (170)](https://github.com/khoj-ai/khoj/assets/77260113/63c000da-11f2-4ba8-a9d9-6029cf7c749b)
![Screenshot (171)](https://github.com/khoj-ai/khoj/assets/77260113/f1aeffa1-8e5b-4a37-a5a7-22823a70c5c8)
![Screenshot (172)](https://github.com/khoj-ai/khoj/assets/77260113/4ee7f26b-9b1d-488e-9b2c-c90c9ec15d06)
![Screenshot (173)](https://github.com/khoj-ai/khoj/assets/77260113/332f24a5-80ac-47a9-98f5-73a84036583b)
![Screenshot (174)](https://github.com/khoj-ai/khoj/assets/77260113/d019a9b6-452d-40ee-9dd2-bbaa9d82cea9)

**Comment by debanjum** [comment]
Hi @Raghav-45 thanks for the proposal! We are thinking of improving the UI and potentially transitioning to NextJS.
Is this chat UI something you've already built (i.e in code) or are these design prototypes? And what help would you need to contribute to the Khoj UI?

**Reply by Raghav-45** [reply]
I have successfully installed the application within a local machine using Docker, utilizing the docker-compose.yaml file. Additionally, I've executed the necessary configurations within the database.

Please find below a screenshot illustrating the steps taken:
![image](https://github.com/khoj-ai/khoj/assets/77260113/2de12e23-79ec-410f-867f-f46859fc76ff)

but it just loads for infinite time:
![image](https://github.com/khoj-ai/khoj/assets/77260113/555405ad-7231-4aec-9c1c-4c448cbe3bc0)

**Reply by debanjum** [reply]
Hey @Raghav-45, Khoj transitioned to using Next.js for the web app. Do try out the latest Khoj to see if it feels better?
